onnxruntime/core/graph/graph.cc

```diff
@@ -2337,6 +2337,13 @@ const std::vector<const NodeArg*>& Graph::GetValueInfo() const noexcept {
   return value_info_;
 }
 
+void Graph::AddValueInfo(const NodeArg* new_value_info){
+  for(auto info : value_info_){
```

Comment:
>auto [](start = 6, length = 4)

const auto&

Reply:
Not sure worth changing it to reference -- as value_info_ is a vector of pointers, and a reference is also a pointer, so taking it by value v.s. by reference would take the same memory space.

I think auto will deduce the 'const', but I think it is a good idea to explicitly put it there. 

---
In reply to: [428256226](https://github.com/microsoft/onnxruntime/pull/3996#discussion_r428256226) [](ancestors = 428256226)

Reply:
ok, better to use conat auto* or auto* for pointer types.

---
In reply to: [428394644](https://github.com/microsoft/onnxruntime/pull/3996#discussion_r428394644) [](ancestors = 428394644,428256226)

Reply:
Updated.

---
In reply to: [428987507](https://github.com/microsoft/onnxruntime/pull/3996#discussion_r428987507) [](ancestors = 428987507,428394644,428256226)